spotify_tensorflow/dataset.py

```diff
@@ -118,8 +99,8 @@ def _get_tfrecord_filenames(dir_path):
         return filenames
 
     @staticmethod
-    def _get_default_feature_mapping_fn(feature_names):
-        fm = [(name, tf.FixedLenFeature((), tf.int64, default_value=0)) for name in feature_names]
+    def _get_default_feature_mapping_fn(feature_info):
+        fm = [(f.name, tf.FixedLenFeature((), f.kind, default_value=0)) for f in feature_info]
```

Comment:
What if the feature is a list.  Do we need to pass in the length here?

Reply:
@richwhitjr  You mean [`tf.VarLenFeature`](https://www.tensorflow.org/api_docs/python/tf/VarLenFeature)? 

In this case this code would not work and the user would need to inject its own `feature_mapping_fn` to specify manually how to parse the features. 
We thought about ways to let the user specify this  information from scala annotations but it seems hairy and seems like an advanced use case (featran does not support VarLenFeature yet.) 
Do you have suggestions?

Reply:
I see that makes sense.  Missed the part where you can override this.  

We could also output feature length with the spec.  Certain Transformers must have a fixed length and we calculate it on the aggregation step so if a settings file is produced we know the length of the vectors.

It would require this code to pull the settings file though.  I believe right now also there is no feature you can't know the length based on the settings file.

Reply:
When you say Transformers, you mean within Featran right? 
 
The way Featran works right now, even if a feature outputs a variable length vector (like a one hot encoding), Featran will generate the right number of features (`my_feature_0`... `my_feature_42`) so it becomes of bunch of FixedLenFeatures in the TFRecord. 

So this code should work with any Featran FeatureSpec - Am I missing a something?

Reply:
Ah I see, so it outputs one float per column not as a list.  So each FixedLenFeature size is 1?  That makes sense.